src/Sulu/Component/Content/Types/Rlp/Strategy/RlpStrategy.php

```diff
@@ -92,11 +101,20 @@ public function getName()
     /**
      * {@inheritdoc}
      */
-    public function generate($title, $parentPath, $webspaceKey, $languageCode, $segmentKey = null)
+    public function generate($title, $parentUuid, $webspaceKey, $languageCode, $segmentKey = null)
     {
         // title should not have a slash
         $title = str_replace('/', '-', $title);
 
+        if ($parentUuid !== null) {
+            $parentDocument = $this->documentManager->find($parentUuid, $languageCode, ['load_ghost_content' => false]);
+            // using loadByContentUuid because loadByContent returns the wrong language for shadow-pages
+            $pathParentUuid = $this->documentInspector->getUuid($this->getPublishedAncestorOrSelf($parentDocument));
```

Comment:
Why not `resolvedParentUuid`? I guess you are just doing this because the given `parentUuid` might lead to a page which is not published yet. Would add that as a comment, and rename the variable, since it has absolutely nothing to do with a path.